index.js

```diff
@@ -25,6 +25,20 @@ function postHTMLParser(html, options) {
         return this[this.length - 1];
     };
 
+    function isDirective(directive, tag) {
+        if (directive.name instanceof RegExp) {
+            var regex = RegExp(directive.name.source, 'i');
```

Comment:
Final nitpick `directive.name.source` => `directive.name` (But both work)

Reply:
travis gives this error if we give regex parameter for regex and supply flag.
```
1) PostHTML-Parser test should be parse regular expression directive:
     TypeError: Cannot supply flags when constructing one RegExp from another
      at new RegExp (native)
      at RegExp (native)
      at isDirective (index.js:9:991)
      at Object.parserDirective [as onprocessinginstruction] (index.js:9:2022)
      at Parser.onprocessinginstruction (node_modules/htmlparser2/lib/Parser.js:275:13)
      at Tokenizer._stateInProcessingInstruction (node_modules/htmlparser2/lib/Tokenizer.js:345:13)
      at Tokenizer._parse (node_modules/htmlparser2/lib/Tokenizer.js:686:9)
      at Tokenizer.write (node_modules/htmlparser2/lib/Tokenizer.js:632:7)
      at Parser.write (node_modules/htmlparser2/lib/Parser.js:334:18)
      at postHTMLParser (index.js:9:5732)
      at parser (index.js:9:6155)
      at parserWrapper (index.js:9:6665)
      at Context.<anonymous> (test/test.js:143:16)
```
I have tried in https://github.com/posthtml/posthtml-parser/commit/f5249b2b9b6f002ba8c68bd600867e5b73b92a7f and the test result https://travis-ci.org/posthtml/posthtml-parser/builds/330561437 is failed.

Reply:
🙇 I misread the MDN snippet which clearly states **starting** from ECMAScript 2015 (6) it no long throws. You're right for the current targets this needs definitely to be handled via `pattern.source`

Reply:
Is there any plan to move to ES6?

Reply:
we support the node> = 0.10